components/mdns/private_include/mdns_private.h

```diff
@@ -212,18 +214,6 @@ typedef enum {
 typedef struct {
     uint16_t id;
     union {
```

Comment:
Nitpick: Could we also remove the union here? (so instead of `header.flags.value` we'd just say `header.flags` ?)

Reply:
(if you're addressing this, could you please rebase as well? we've just merged some fixes to CI failures against IDF master)